File: frontend/src/microservices/traveler/objects/TravelerDocumentInfo.ts
// 本文件定义 traveler 模块的 `TravelerDocumentInfo`，作为证件传输数据并提供 JSON 编解码。

export type TravelerDocumentInfo = {
  documentType: string
  documentNumber: string
  documentExpiryDate: string | null
}

export const travelerDocumentInfoFromJson = (json: string): TravelerDocumentInfo =>
  JSON.parse(json) as TravelerDocumentInfo

export const travelerDocumentInfoToJson = (value: TravelerDocumentInfo): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/traveler/objects/TravelerDocumentInfo.scala
// 本文件定义 traveler 模块的证件传输对象 `TravelerDocumentInfo`，仅负责数据结构与 JSON codec。
package com.typesafe.travel.traveler.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TravelerDocumentInfo(
    documentType: String,
    documentNumber: String,
    documentExpiryDate: Option[String]
)

object TravelerDocumentInfo:
  given sourceEncoder: Encoder[TravelerDocumentInfo] = deriveEncoder
  given sourceDecoder: Decoder[TravelerDocumentInfo] = deriveDecoder
